web/main.py

```diff
@@ -43,17 +62,32 @@ def handler(dataset=None, filename=None):
         logger.warning(f'Invalid dataset "{dataset}"')
         abort(400)
 
+    bucket_name = f'cpg-{dataset}-{BUCKET_SUFFIX}'
+    bucket = storage_client.bucket(bucket_name)
+
     dataset_project_id = dataset_config['projectId']
     members = read_secret(
         dataset_project_id, f'{dataset}-web-access-members-cache'
     ).split(',')
+
     if email not in members:
-        logger.warning(f'{email} is not a member of the {dataset} access group')
-        abort(403)
+        # Second chance: if there's a '.access' file in the first subdirectory,
+        # check if the email is listed there.
+        split_subdir = filename.split('/', maxsplit=1)
```

Comment:
I think using os.path.dirname is cleaner for the downstream code
```suggestion
        subdir = os.path.dirname(filename)
```

Reply:
I only want the first subdirectory down from the top-level though, not the `.access` file further down. I thought about allowing those files at any subdirectory level, but then you'd have to walk up the tree and check for blob existence on every level.